c7n/resources/emr.py

```diff
@@ -307,6 +307,50 @@ def get_related_ids(self, resources):
 filters.register('network-location', net_filters.NetworkLocation)
 
 
+@filters.register('security-configuration')
+class EMRSecurityConfigurationFilter(ValueFilter):
+    """Filter for annotate security configuration and
+       filter based on its attributes.
+
+    :example:
+
+    .. code-block:: yaml
+
+      policies:
+        - name: emr-security-configuration
+          resource: emr
+          filters:
+            - type: security-configuration
+              key: EnableAtRestEncryption
+              value: true
+
+    """
+    annotation_key = 'c7n:SecurityConfiguration'
+    permissions = ("elasticmapreduce:ListSecurityConfigurations",
+                   "elasticmapreduce:DescribeSecurityConfiguration",)
+    schema = type_schema('security-configuration', rinherit=ValueFilter.schema)
+    schema_alias = False
+
+    def process(self, resources, event=None):
+        results = []
+        emr_sec_cfgs = self.manager.get_resource_manager(
+            'emr-security-configuration').resources()
+        for r in resources:
+            if self.annotation_key not in r:
+                r[self.annotation_key] = dict()
+            if 'SecurityConfiguration' in r:
+                config_name = r['SecurityConfiguration']
+                for emr_sec_cfg in emr_sec_cfgs:
+                    if config_name in emr_sec_cfg['Name']:
+                        r[self.annotation_key] = \
+                            emr_sec_cfg['SecurityConfiguration']['EncryptionConfiguration']
```

Comment:
security configuration covers three components, encryption, authentication, authorization. it feels strange that we would omit the other two.